internal/aez/ccd_data.go

```diff
@@ -0,0 +1,169 @@
+package aez
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/innosat-mats/rac-extract-payload/internal/ccsds"
+)
+
+// WDWMode describes the CCD WDW parameter
+type WDWMode uint8
+
+// WDWModeManual = 0, check WDWOV for overflows
+//
+// WDWModeAutomatic, window selected depending on input data
+const (
+	WDWModeManual    WDWMode = 0
+	WDWModeAutomatic WDWMode = 1
+)
+
+// Wdw is the composite status of Image Window Mode
+type Wdw uint8
+
+// Mode returns the WDWMode type used encoded in Bit[7]
+func (wdw Wdw) Mode() WDWMode {
+	if (wdw & 0b0000001) == 0 {
+		return WDWModeManual
+	}
+	return WDWModeAutomatic
+}
+
+// InputDataWindow returns which bits out of the depth of the CCD Images
+//
+// It ueses information in Bit[2..0]
+//
+// Returns major and minor bit of original CCD Image and error
+// if encoding not covered by specification.
+//
+// If full range of Bit[15..0] is used the JPEGQ should be 0xFF
+func (wdw Wdw) InputDataWindow() (int, int, error) {
+	switch wdw & 0b111 {
+	case 0x0:
+		return 11, 0, nil
+	case 0x1:
+		return 12, 1, nil
+	case 0x2:
+		return 13, 2, nil
+	case 0x3:
+		return 14, 3, nil
+	case 0x4:
+		return 15, 4, nil
+	case 0x7:
+		return 15, 0, nil
+	}
+	return -1, -1, fmt.Errorf(
+		"WDW value has unknown Input Data Window '%x'", wdw&0b111,
+	)
+}
+
+// NCBin contains the FPGA and CCD columns bin count
+type NCBin uint16
+
+// FPGAColumns returns number FPGA columns to bin, Bit[11..8]
+func (ncBin NCBin) FPGAColumns() int {
+	return 1 << ((ncBin >> 8) & 0x0f)
+}
+
+// CCDColumns returns number of CCD columns to bin, Bit[7..0]
+func (ncBin NCBin) CCDColumns() int {
+	return (int)(ncBin & 0xff)
+}
+
+// CCDGain is game composite information
+type CCDGain uint16
+
+// CCDGainMode is high/low signal mode
+type CCDGainMode int
+
+// HighSignalMode, LowSignalMode
+const (
+	HighSignalMode CCDGainMode = iota
+	LowSignalMode
+)
+
+// CCDGainTiming is the timing flag
+type CCDGainTiming int
+
+// FasterTiming = 0, used for binned and discarded pixels
+//
+// FullTiming = 1, used even for pixels that are not read out
+const (
+	FasterTiming CCDGainTiming = iota
+	FullTiming
+)
+
+// Mode returns high/low signal mode, Bit[12]
+func (gain CCDGain) Mode() CCDGainMode {
+	if (gain & 0b00000000000001) == 0 {

```

Comment:
mask for bit 12: 0b00010000 00000000 or 0x1000

Reply:
In my experience, long bit masks quickly get unreadable. Though it requires some getting used to, I believe hex-masks are considered better practice.

Reply:
I think i could manage `f`,`2`,`1`, and `0` values for hexmasks but else you are asking me to make mistakes

Reply:
I'm asking you to learn not to make mistakes! ;)

Reply:
I certainly won't insist, but if I'm to work with similar problems in the code, I would prefer to use hex-based masks, for the same reason you prefer bit-masks. This comment is only an excuse to share this link, btw: https://flippybitandtheattackofthehexadecimalsfrombase16.com/
:-D

Reply:
cool, but I kind of wonder if they learn bit to hex in a sense we would understand it or if they learn a pattern to enemy position equals reward.